lib/components/rtsp-session/rtspsession-component.js

```diff
@@ -157,7 +161,23 @@ class RtspSessionComponent extends Component {
     if (!this._sessionId && !ended) {
       // Response on first SETUP
       this._sessionId = Rtsp.sessionId(msg.data)
+      const sessionTimeout = Rtsp.sessionTimeout(msg.data)
+      if (sessionTimeout !== null) {
+        // The server specified that sessions will timeout if not renewed.
+        // In order to keep it alive we need periodically send a RTSP_OPTIONS message
+        if (this._renewSessionInterval) {
+          clearInterval(this._renewSessionInterval)
+        }
+        this._renewSessionInterval = setInterval(
+          () => {
+            this._enqueue({ method: RTSP_OPTIONS })
+            this._dequeue()
+          },
+          (sessionTimeout - 5) * 1000 // timeout minus 5 seconds.
```

Comment:
I wonder if we need a lower limit here?
In any case, I would put a guard like e.g.
```
const MIN_SESSION_TIMEOUT = 5 // somewhere at top of file
...
Math.max(MIN_SESSION_TIMEOUT, sessionTimeout - 5) * 1000
```

Reply:
Absolutly agreed